smartcard/pcsc/pcsclite_client.go

```diff
@@ -213,8 +213,8 @@ func (client *PCSCLiteClient) Write(data []byte) (int, error) {
     return client.connection.Write(data)
 }
 
-func (client *PCSCLiteClient) EstablishContext() (uint32, error) {
-    estruct := establishStruct{scope: CARD_SCOPE_SYSTEM}
+func (client *PCSCLiteClient) EstablishContext(scope uint32) (uint32, error) {
```

Comment:
Change method signature to use variadic parameter, which effectively makes it optional. Use SCOPE_SYSTEM as default

`func (client *PCSCLiteClient) EstablishContext(scope ...uint32) (uint32, error) {`

See [this stack overflow post](https://stackoverflow.com/questions/2032149/optional-parameters-in-go/19813113#19813113) for reference

Reply:
Done, thanks for the idea!